senders/telegram/send.go

```diff
@@ -31,6 +33,30 @@ var characterLimits = map[messageType]int{
 	Album:   albumCaptionMaxCharacters,
 }
 
+type Chat struct {
+	ID       int64            `json:"chatId" example:"-1001234567890"`
+	Type     telebot.ChatType `json:"type" example:"supergroup"`
+	ThreadID int              `json:"threadId,omitempty" example:"10"`
+}
+
+var brokenContactAPIErrors = map[*telebot.Error]bool{
```

Comment:
It's good to put this in a separate variable, you could also change the type to `map[*telebot.Error]struct{}`, because we don't care about the value

Reply:
Done https://github.com/moira-alert/moira/pull/1022/commits/57f1b7104a24527fa9ce343fc492bb872a33d44f

NB: I think it would be better to use a slice of errors and `slices.Contains` method introduced in go 1.21 since project has recently updated to 1.21. But this changes are not merged into this PR yet.

<details><summary>Code snipplet</summary>
<p>

```go
import (
	"slices"
)

...

var brokenContactAPIErrors = []*telebot.Error{
	telebot.ErrUnauthorized,
	telebot.ErrUserIsDeactivated,
	telebot.ErrNoRightsToSendPhoto,
	telebot.ErrChatNotFound,
	telebot.ErrNoRightsToSend,
	telebot.ErrKickedFromGroup,
	telebot.ErrBlockedByUser,
	telebot.ErrKickedFromSuperGroup,
	telebot.ErrKickedFromChannel,
	telebot.ErrNotStartedByUser,
}

...

func isBrokenContactAPIError(err *telebot.Error) bool {
	return slices.Contains(brokenContactAPIErrors, err)
}

```

</p>
</details>

Reply:
Checking in `slices.Contains` has O(N) asymptotics while checking in map for amortized O(1). In this case, given that there are few elements, `slices.Contains` may be even faster, but I would leave it as it is